### PR TITLE
Throw an error if > 1 folder is found per gene ID at the FTP site

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: src/loaders/ncbi_downloader/ncbi_downloader_helper.py [hunk 1]
```diff
@@ -36,7 +36,10 @@ def _download_genome_file(
 
                 host.chdir(gene_dir)
                 dir_list = host.listdir(host.curdir)
-                gene_dir_name = [i for i in dir_list if i.startswith(gene_id)][0]
+                gene_dir_list = [i for i in dir_list if i.startswith(gene_id)]
+                if len(gene_dir_list) > 1:
```
Comment (MrCreosote) [inline src/loaders/ncbi_downloader/ncbi_downloader_helper.py:40]: Might as well just do `== 1` and make sure there's at least 1 dir as well

Comment (Xiangs18) [reply src/loaders/ncbi_downloader/ncbi_downloader_helper.py]: Updated!

**Review by Tianhao-Gu — DISMISSED** [review]
👍

File: src/loaders/ncbi_downloader/ncbi_downloader_helper.py [hunk 2]
```diff
@@ -36,7 +36,11 @@ def _download_genome_file(
 
                 host.chdir(gene_dir)
                 dir_list = host.listdir(host.curdir)
-                gene_dir_name = [i for i in dir_list if i.startswith(gene_id)][0]
+                gene_dir_list = [i for i in dir_list if i.startswith(gene_id)]
+                if len(gene_dir_list) != 1:
+                    raise ValueError(f"One and only one directory starts with {gene_id} "
```
Comment (MrCreosote) [inline src/loaders/ncbi_downloader/ncbi_downloader_helper.py:41]:
```suggestion
                    raise ValueError(f"One and only one directory that starts with {gene_id} "
```

**Review by MrCreosote — APPROVED** [review]

File: src/loaders/ncbi_downloader/ncbi_downloader_helper.py [hunk 3]
```diff
@@ -36,7 +36,11 @@ def _download_genome_file(
...
+                                     f"must be present in https://ftp.ncbi.nlm.nih.gov{gene_dir}")
```
Comment (MrCreosote) [inline src/loaders/ncbi_downloader/ncbi_downloader_helper.py:42]:
```suggestion
                                     f"must be present in https://{ncbi_domain}{gene_dir}")
```

File: src/loaders/ncbi_downloader/ncbi_downloader_helper.py (hunk 2 above)
Comment (Xiangs18) [reply src/loaders/ncbi_downloader/ncbi_downloader_helper.py]: Updated!

File: src/loaders/ncbi_downloader/ncbi_downloader_helper.py (hunk 3 above)
Comment (Xiangs18) [reply src/loaders/ncbi_downloader/ncbi_downloader_helper.py]: Updated!

**Review by MrCreosote — APPROVED** [review]

**Review by Tianhao-Gu — APPROVED** [review]
LGTM

**Pull request merged**